packages/cdktf-cli/src/bin/cmds/helper/terraform-cloud-client.ts

```diff
@@ -39,6 +39,49 @@ export interface Workspaces {
   data: WorkspaceData[];
 }
 
+export interface ServiceDiscovery {
+  "modules.v1": string;
+  "providers.v1": string;
+  "motd.v1": string;
+  "state.v2": string;
+  "tfe.v2": string;
+  "tfe.v2.1": string;
+  "tfe.v2.2": string;
+  "versions.v1": string;
+}
+
+let cachedServiceDiscovery: ServiceDiscovery | undefined;
+async function discoverService(hostname: string): Promise<ServiceDiscovery> {
+  if (cachedServiceDiscovery) return Promise.resolve(cachedServiceDiscovery);
```

Comment:
nit: You don't need `Promise.resolve` in an async function
```suggestion
  if (cachedServiceDiscovery) return cachedServiceDiscovery;
```